core/deployment/src/main/java/io/quarkus/deployment/configuration/ConfigMappingUtils.java

```diff
@@ -109,7 +112,12 @@ static void processConfigClass(
         // For implicit converters
         for (ClassInfo classInfo : mappingsInfo) {
             for (MethodInfo method : classInfo.methods()) {
-                reflectiveClasses.produce(new ReflectiveClassBuildItem(true, false, method.returnType().name().toString()));
+                Type type = method.returnType();
+                if (type.name().equals(OPTIONAL)) {
```

Comment:
@radcortez this should fix the problem with `Optional`. It's not a robust solution but I have no idea whether it's needed for other containers (such as `List`).

Reply:
Hum, yes, we probably need it for List as well.

Reply:
How about `Set` and others? Should I file a separate issue?

Reply:
https://github.com/quarkusio/quarkus/issues/28065